commcare-cloud-bootstrap/commcare_cloud_bootstrap.py

```diff
@@ -90,14 +100,17 @@ class Group(StrictJsonObject):
     vars = jsonobject.DictProperty()
 
 
-def provision_machines(spec, env_name=None):
+def provision_machines(spec, env_name=None, create_machines=True):
     if env_name is None:
         env_name = u'hq-{}'.format(
             ''.join(random.choice(string.ascii_lowercase + string.digits) for _ in range(7))
         )
     environment = get_environment(env_name)
     inventory = bootstrap_inventory(spec, env_name)
-    instance_ids = ask_aws_for_instances(env_name, spec.aws_config, len(inventory.all_hosts))
+    if create_machines:
+        instance_ids = ask_aws_for_instances(env_name, spec.aws_config, len(inventory.all_hosts))
```

Comment:
It seems odd to me that a function called `provision_machines` in a module named `commcare_cloud_bootstrap`, both fairly generic and could be expected to work in many hosting contexts, would make specific references to AWS. Is it expected that this module will only ever be used to bootstrap commcare clouds on Amazon servers?

I see there were references to AWS here before this PR, so this comment probably does not apply only to the changes being made here.

Thinking out loud a bit more: I suppose we're writing this for AWS first, and then maybe someday if we ever use it on some hosting provider other than Amazon we'll abstract out the references to AWS around a more generic API. Is it true that right now it's too early to know what the generic API would look like, and so it's easier to just reference AWS even though it's overly specific?

Reply:
Yeah, so right now this is set up only for me and Preethi to use for what we're doing right now, and is intentionally not being presented as a more general purpose tool. Some of the things it does are sort of test beds for actual tools we'll probably want to build, such as the ability to generate environment files with reasonable presets, etc.; when we want to roll out those tools, we'll probably pull out code from here and put it somewhere appropriate and accessible, document it, etc.

For now, this is code for automating a particular setup involving AWS. Depending on what we actually use it for in the future, I'm sure it'll morph into something else, but I didn't want to code to a general problem that we're not yet facing, and am not even sure we'll stick with this versus replacing some of it with a commonly used tool like terraforms or something.

Reply:
> I didn't want to code to a general problem that we're not yet facing

👍 makes sense